src/Microsoft.OData.Core/UriParser/SemanticAst/EachSegment.cs

```diff
@@ -0,0 +1,90 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="EachSegment.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.UriParser
+{
+    #region Namespaces
+    using Microsoft.OData.Edm;
+    #endregion Namespaces
+
+    /// <summary>
+    /// A segment representing $each in a path.
+    /// </summary>
+    public sealed class EachSegment : ODataPathSegment
+    {
+        /// <summary>
+        /// The <see cref="IEdmType"/> of the resource that this <see cref="EachSegment"/> represents.
+        /// </summary>
+        private readonly IEdmType edmType;
+
+        /// <summary>
+        /// Build a segment representing each.
+        /// </summary>
+        /// <param name="navigationSource">The entity collection that this set-based operation applies to.</param>
+        /// <param name="targetEdmType">Target type for the entity being referenced.</param>
+        /// <exception cref="System.ArgumentNullException">Throws if any input parameter is null.</exception>
+        /// <remarks>$filter cannot be applied on singletons.</remarks>

```

Comment:
>filter [](start = 22, length = 6)

$each

Reply:
:+1:

Reply:
Doesn't appear to be fixed?

---
In reply to: [224584557](https://github.com/OData/odata.net/pull/1110#discussion_r224584557) [](ancestors = 224584557)

Reply:
Odd...I distinctly remember fixing this as well as the other code comment ones. Actually doing it right now. Marking unresolved for now.